src/main/java/org/truffleruby/core/rubinius/IONodes.java

```diff
@@ -805,4 +753,32 @@ public DynamicObject setLastLine(VirtualFrame frame, DynamicObject matchData) {
         }
     }
 
+    @Primitive(name = "io_get_thread_buffer", needsSelf = false)
+    public static abstract class GetThreadBufferNode extends PrimitiveArrayArgumentsNode {
+
+        private static final ThreadLocal<Pointer> THREAD_BUFFER_THREAD_LOCAL = ThreadLocal.withInitial(() -> Pointer.NULL);
```

Comment:
We should free() this when a Thread dies.

Reply:
Right - I think I'll just use an autorelease for safety.